### PR TITLE
First commit

### PR DESCRIPTION
**Pull request by mafintosh**
Been working pretty well so far.

You need a Kylin testnet account + faucet to test it.
This is my current test code:

```js
const deos = require('./')

/*
const { pay } = deos({
  account: 'mafintoshmaf',
  privateKey: '...'
})
`
pay('mafinotshcpy', '0.5678 EOS', 'dazaar key', function (err) {
  console.log('payed', err)
})
*/

const { subscription } = deos({
 account: 'mafinotshcpy'
})

const s = subscription('dazaar key', '0.001 EOS/s')

console.log(s.active())

s.on('update', function () {
  console.log(s.active())
})
```

### CODE REVIEW

**Comment by mafintosh** [comment]
Needs review from @emilbayes and @robertkowalski :)

**Comment by mafintosh** [comment]
Create an account using this link

http://faucet.cryptokylin.io/create_account?{add-12-char-account-name-here}

And get coins using this faucet

http://faucet.cryptokylin.io/get_token?{valid_account_name}

File: index.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    subscription,
+    createTransactionStream
+  }
+
+  function pay (destination, amount, memo, cb) {
+    if (!api) throw new Error('opts.privateKey must be provided in the constructor')
+
+    api.transact({
+      actions: [{
+        account: 'eosio.token',
```
Comment (robertkowalski) [inline index.js]: for other tokens, other contracts are used. (e.g. tether on eos)

maybe make it configurable

File: index.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        name: 'transfer',
+        authorization: [{
+          actor: account,
+          permission: 'owner'
```
Comment (robertkowalski) [inline index.js]: owner permission is like root - usually active is used for transfers

File: index.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    const perSecond = Number(match[1])
+    const sub = new EventEmitter()
+
+    const stream = createTransactionStream()
+    const activePayments = []
+
+    stream.on('data', function (data) {
+      if (data.act.data.memo !== filter) return
+      if (data.act.data.to !== account) return
```
Comment (robertkowalski) [inline index.js:76]: is the tx already irreversibe?

take a look here for verifying incoming txs:

```
actions[0].action_trace.act.account == "eosio.token" &&
    actions[0].action_trace.act.name == "transfer" &&
    actions[0].action_trace.act.data.quantity == "X.0000 SYS" &&
    actions[0].action_trace.to == "tokenxchange" && 
    actions[0].action_trace.memo == "KEY TO IDENTIFY INTERNAL ACCOUNT" && 
    actions[0].action_trace.receipt.receiver == "tokenxchange"  &&
    actions[0].block_num < last_irreversible_block
```

(from https://developers.eos.io/eosio-cpp/docs/exchange-deposit-withdraw)

**Review by robertkowalski — CHANGES_REQUESTED** [review]
looks nice, have a few questions

File: index.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    function destroy (err) {
+      stream.destroy(err)
+    }
+  }
+}
+
+function parseQuantity (s) {
+  const m = s.match(/^(\d+(?:\.\d+)?) EOS$/)
+  if (!m) return 0
+  return Number(m[1])
```
Comment (robertkowalski) [inline index.js:171]: probably use bignum.js or similar makes sense here

File: index.js (hunk 2 above)
Comment (mafintosh) [reply index.js]: i'll change it. should be configurable also i'm guessing?

File: index.js (hunk 1 above)
Comment (mafintosh) [reply index.js]: 👍 silly question: does the quantity format change depending on which contract is used?

File: index.js (hunk 4 above)
Comment (mafintosh) [reply index.js:171]: 👍

File: index.js (hunk 3 above)
Comment (mafintosh) [reply index.js:76]: ah nice, didn't know :)

File: index.js (hunk 1 above)
Comment (robertkowalski) [reply index.js]: yes, it can :(

File: index.js (hunk 2 above)
Comment (robertkowalski) [reply index.js]: yes, that would be cool.

File: index.js (hunk 3 above)
Comment (mafintosh) [reply index.js:76]: fixed in latest

File: index.js (hunk 4 above)
Comment (mafintosh) [reply index.js:171]: Will hit this in another pr :)

**Comment by mafintosh** [comment]
@prdn ready to merge

**Pull request merged**